database/src/main/postgres/flows/V1.9.1__get_flow_checkpoints_v2.sql

```diff
@@ -66,53 +70,60 @@ $$
 --      measured_columns       - measure columns associated with a given checkpoint
 --      measurement_value      - measurement details associated with a given checkpoint
 --      checkpoint_time        - time
+--      has_more               - flag indicating whether there are more checkpoints available
 --
 -- Status codes:
---      11                  - OK
---      41                  - Partitioning not found
---
--------------------------------------------------------------------------------
-
+--      11                     - OK
+--      42                     - Flow not found
+---------------------------------------------------------------------------------------------------
 DECLARE
-    _fk_partitioning BIGINT;
-    _fk_flow BIGINT;
+    _actual_limit INT := i_limit + 1;
+    _record_count INT;
 BEGIN
-    _fk_partitioning = runs._get_id_partitioning(i_partitioning_of_flow);
+    -- Execute the query to retrieve checkpoints flow and their associated measurements
+    RETURN QUERY
+        SELECT 11 AS status,
+               'OK' AS status_text,
+               CP.id_checkpoint,
+               CP.checkpoint_name,
+               CP.created_by AS author,
+               CP.measured_by_atum_agent,
+               MD.measure_name,
+               MD.measured_columns,
+               M.measurement_value,
+               CP.process_start_time AS checkpoint_start_time,
+               CP.process_end_time AS checkpoint_end_time,
+               (ROW_NUMBER() OVER ()) <= i_limit AS has_more
+        FROM flows.partitioning_to_flow AS PF
+                 JOIN runs.checkpoints AS CP
+                      ON PF.fk_partitioning = CP.fk_partitioning
+                 JOIN runs.measurements AS M
+                      ON CP.id_checkpoint = M.fk_checkpoint
+                 JOIN runs.measure_definitions AS MD
+                      ON M.fk_measure_definition = MD.id_measure_definition
+        WHERE PF.fk_flow = i_flow_id
+          AND (i_checkpoint_name IS NULL OR CP.checkpoint_name = i_checkpoint_name)
+        ORDER BY CP.process_start_time,
+                 CP.id_checkpoint
+        LIMIT _actual_limit
+        OFFSET i_offset;
+
+    GET DIAGNOSTICS _record_count = ROW_COUNT;
+
+    IF _record_count > i_limit THEN
+        has_more := TRUE;
+    ELSE
+        has_more := FALSE;
+    END IF;
 
-    IF _fk_partitioning IS NULL THEN
-        status := 41;
-        status_text := 'Partitioning not found';
+    IF NOT FOUND THEN
```

Comment:
Also rather strange, the fact that the return query does return records or not does not imply if flow was or wasn't found.

Reply:
Will fix it.